api/include/opentelemetry/context/context.h

```diff
@@ -0,0 +1,83 @@
+#pragma once
+
+
+#include "opentelemetry/common/attribute_value.h"
+#include "opentelemetry/nostd/string_view.h"
+#include "opentelemetry/trace/key_value_iterable_view.h"
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace context
+{
+
+// The context class provides a context identifier.
+// This is a dummy class that is meant to be overridden,
+// the methods return default values.
+class Context
+{
+
+public:
+  // The Key class is used to obscure access from the
+  //  user to the context map. The identifier is used as a key
+  // to the context map.
+  
+  class Key
+  {
+
+  public:
+    // Returns the key's identifier.
+    nostd::string_view GetIdentifier(){
+      return "";
+    }
+    
+    // Returns the key's name
+    nostd::string_view GetName(){
+      return "";
+    }
+
+  private:
+    friend class Context;
+
+    // Constructs a new Key with the passed in name. Sets the identifier as
+    // the address of this object.
+    Key(nostd::string_view key_name){}
+
+    char key_name_[50];
+
+    char identifier_[50];
+  };
+
+  // Creates a key with the passed in name and returns it. 
+  Key CreateKey(nostd::string_view key_name){
+    return Key(key_name);
+  }
+
+  Context(){}
```

Comment:
Google style is to use = default, but IDK if Open Telemetry has a preference
 
```suggestion
  Context() = default.
```